zne/noise_amplification/folding_amplifier/glorious_local_folding_amplifier.py

```diff
@@ -0,0 +1,170 @@
+            dag.apply_operation_back(inverted_node, qargs=node.qargs, cargs=node.cargs)
+            dag.apply_operation_back(barrier, qargs=node.qargs)
+            dag.apply_operation_back(node.op, qargs=node.qargs, cargs=node.cargs)
+        dag.apply_operation_back(barrier, qargs=node.qargs)
+        return dag
+
+    def _compute_num_foldings(self, noise_factor: float) -> int:
+        """Compute number of foldings.
+
+        Args:
+            noise_factor (float) : The original noise_factor input.
+
+        Returns:
+            int: Number of foldings calculated from noise_factor.
+        """
+        noise_factor = self._validate_noise_factor(noise_factor)
+        return int((noise_factor - 1) / 2)
+
+    def _validate_gates_to_fold(self, gates_to_fold: Sequence[int | str] | None) -> None:
+        """Validates if gates_to_fold is valid.
+
+        Args:
+            gates_to_fold (Sequence[int | str] | None) : Original gates_to_fold input.
+        """
+        if isinstance(gates_to_fold, int):
+            return
+        gate_list = list(standard_gates.get_standard_gate_name_mapping().keys())
+        for value in gates_to_fold:
+            if value not in gate_list:
+                try:
+                    int(value)
+                except ValueError:
+                    raise ValueError(  # pylint: disable=raise-missing-from
+                        f"{value} not a valid gate or value."
+                    )
+
+    # pylint: disable=duplicate-code
+
+    def _validate_noise_factor(self, noise_factor: float) -> float:
+        """Normalizes and validates noise factor.
+
+        Args:
+            noise_factor (float) : The original noisefactor input.
+
+        Returns:
+            float: Normalised noisefactor input.
+
+        Raises:
+            ValueError: If input noise_factor value is not of type float.
+            TypeError: If input noise_factor value is not of type float.
+        """
+        try:
+            noise_factor = float(noise_factor)
+        except ValueError:
+            raise ValueError(  # pylint: disable=raise-missing-from
+                f"Function call expects a positive floating value. "
+                f"Received value of {noise_factor} instead."
+            )
+        except TypeError:
+            raise TypeError(  # pylint: disable=raise-missing-from
+                f"Function call expects a positive floating value. "
+                f"Received value of {noise_factor} instead."
+            )
+        if noise_factor < 1:
+            raise ValueError(  # pylint: disable=raise-missing-from
+                f"Function call expects a positive float noise_factor >= 1."
+                f"Received {noise_factor} instead."
+            )
+        if noise_factor % 2 == 0:
+            raise ValueError(  # pylint: disable=raise-missing-from
+                f"Function call expects a positive odd noise_factor. "
+                f"Received {noise_factor} instead."
```

Comment:
These errors are too verbose, see my comment on the global amplifier for a more concise way to say this 🙂